VTCareService/src/main/java/edu/vt/cs/vtcare/vtcareservice/api/VTCareResource.java

```diff
@@ -20,4 +23,30 @@ public Provider createProvider(Provider provider) throws Exception {
             throw new Exception("An exception occurred while persisting the provider details.");
         }
     }
+
+    @POST
+    @Path("login")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response login(String json) throws Exception {
+        JsonObject o = new Gson().fromJson(json, JsonObject.class);
+        try {
+            String email = o.get("email").getAsString();
+            String password = o.get("password").getAsString();
+            boolean isProviderLogin = o.get("isProviderLogin").getAsBoolean();
+
+            LoginService loginService = new LoginService();
+            int id = loginService.login(email, password,
+                    isProviderLogin);
+
+            if(id > 0){
+                return Response.ok(id).build();
+            }
+            else {
+                return Response.status(Response.Status.UNAUTHORIZED).build();
```

Comment:
It's a wrong usage of the UNAUTHORIZED enum. It should be used when the 'API Authentication' failed itself. Remember, the api authentication is NOT failing here. It's just that the credentials provided didn't match. So, instead send something maybe a string or boolean value that indicates the result.

Reply:
Actually unauthorized is a perfect way to cater to invalid credentials. It is a standard as seen here as well 
https://auth0.com/blog/forbidden-unauthorized-http-status-codes/
Please do some R&D before making suggestions.

Reply:
It clearly mentions in the doc "401 Unauthorized is the status code to return when the client provides no credentials or invalid credentials.". Please read it carefully.